jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/pinning/ConnectionPinningManager.kt

```diff
@@ -152,6 +155,11 @@ class DefaultConnectionPinningManager(private val project: Project) :
                 } else {
                     UiTelemetry.click(project, "connection_multiple_auths_no")
                 }
+                connectionManager.setKeepCodeWhispererConnection(this)
+                // if new connection is bearerToken, ping it to the current project no matter the response
+                if (newConnection is AwsBearerTokenConnection) {
+                    setPinnedConnection(CodeWhispererConnection.getInstance(), newConnection)
+                }
```

Comment:
why would we ignore the user response?

Reply:
so if user switching from IAM to SSO, although they answer no in the prompt, their current window should remain connected although the new windows should not? This also ties to the above comment for adding that additional check

Reply:
if they switch back to IAM it will ask them again?

Reply:
yes, unless they previously said "do not ask again"

Reply:
I noticed that if they answer no and select do not ask again it will still ask, not sure if it's a feature or bug

Reply:
I think there is no need to overcomplicated the scenarios as it's already complicated, if user has already added BID/SSO in current window we should assume that they want this connection unless they specifically say so, if they have do not ask checked again they can also just log out from the current window it's relatively easy to do from their side too.